test/unit_test/engine/test_engine.py

```diff
@@ -212,10 +214,13 @@ def test_pass_exception(self, caplog):
             model = PyTorchModel(model_loader=pytorch_model_loader, model_path=None)
 
             # execute
-            engine.run(model)
+            output_dir = Path(tempfile.TemporaryDirectory().name)
```

Comment:
looks like this is not actually using a temporary directory? You get a tempfile name but the actual tempfile gets deleted right away since there is no context manager or there is no reference to the temporary directory object 

It is not an issue since `output_dir` gets cleaned up manually. But will it reach the clean up step at 223 if the assert step fails?

Reply:
if you keep a reference to the actual object 
```
temp_dir = tempfile.TemporaryDirectory()
output_dir = Path(tempfile.name)
``` 

It will clean up by itself when it goes out of scope. Or you can manually clean up at the end to be safe `temp_dir.cleanup()`

Reply:
Good catche. Updated.